modules/nlu/src/backend/api.ts

```diff
@@ -205,6 +219,27 @@ export default async (bp: typeof sdk, state: NLUState) => {
     const { botId, id } = req.params
     const botEngine = state.nluByBot[botId].engine1 as ScopedEngine
     await botEngine.storage.deleteEntity(id)
+
+    const intents = await botEngine.storage.getIntents()
+    const awaitsList = intents.map(async intent => {
+      const entitySlotIndex = intent.slots.findIndex(slot => slot.entities.includes(id))
+      if (entitySlotIndex === -1) {
+        return
+      }
+
+      const currentSlot = intent.slots[entitySlotIndex]
+      currentSlot.entities.splice(currentSlot.entities.indexOf(id), 1)
+
+      if (!currentSlot.entities.length) {
```

Comment:
I think slot should not be removed if no entities left in the slot entities, we should assign any. What do you guys think ? @asashour @demogoran @slvnperron

Reply:
Well, I may do it, but it's not too obvious for user, I think. BTW, it's not a big deal to implement it.

Reply:
@Eff: that's a good question. There is already a way for the user to delete `slot`. And deleting it removes all its usage in `intent.utterances`.

On the other hand, slots can't be left empty, without entities, and changing a slot entities from the old entity to `system.any` without the user knowing is not good.

After some thought, I would leave that to the user, i.e. I would show a dialog, that removing this entity would delete A,B,C slots, would you like to remove them, or replace with `X` other entity.

But honestly, I wouldn't push the PR requester to do that. If he likes it, then please go ahead, otherwise we do it in another PR.

Reply:
Right. Agreed

Reply:
Ye, it could be better to move it to another PR.